crates/sui-types/src/gas_model/gas_v2.rs

```diff
@@ -365,19 +370,31 @@ mod checked {
             &mut self.gas_status
         }
 
-        fn bucketize_computation(&mut self) -> Result<(), ExecutionError> {
+        fn bucketize_computation(&mut self, aborted: Option<bool>) -> Result<(), ExecutionError> {
             let gas_used = self.gas_status.gas_used_pre_gas_price();
+            let effective_gas_price =
+                if self.cost_table.max_gas_price_aborted_transactions.is_some()
+                    && aborted.unwrap_or(false)
+                {
+                    // For Move aborts (business logic failures), cap at max but don't exceed user's price
+                    // This minimizes the risk of competing for priority execution in the case that the txn may be aborted.
+                    self.gas_price
+                        .min(self.cost_table.max_gas_price_aborted_transactions.unwrap())
```

Comment:
it will take a little bit more plumbing but I think we should express the max price as a factor of RGP, which you can get from `epoch_store.reference_gas_price()`